src/Lachain.Networking/Hub/ClientWorker.cs

```diff
@@ -61,74 +62,95 @@ public void Start()
-                    Logger.LogTrace(
-                        $"Sending {toSend.Messages.Count} messages to hub, {megaBatchBytes.Length} bytes total, peer = {PeerPublicKey.ToHex()}"); 
-                    var messageTypes = toSend.Messages.Select(m =>
-                        m.MessageCase != NetworkMessage.MessageOneofCase.ConsensusMessage
-                            ? m.MessageCase.ToString()
-                            : m.ConsensusMessage.PrettyTypeString()
-                    );
-                    Logger.LogTrace($"Messages types: {string.Join("; ", messageTypes)}");
-                    foreach (var message in toSend.Messages)
+                    if (toSend.Messages.Count > 0)
                     {
-                        MessageCounter
-                            .WithLabels(PeerPublicKey.ToHex(), message.MessageCase.ToString())
-                            .Inc();
-                        MessageBytesCounter
-                            .WithLabels(PeerPublicKey.ToHex(), message.MessageCase.ToString())
-                            .Inc(message.CalculateSize());
+                        var megaBatch = _messageFactory.MessagesBatch(toSend.Messages);
+                        var megaBatchBytes = megaBatch.ToByteArray();
+                        if (megaBatchBytes.Length == 0)
+                            throw new Exception("Cannot send empty message");
+                        Logger.LogTrace(
+                            $"Sending {toSend.Messages.Count} messages to hub, {megaBatchBytes.Length} bytes total, peer = {PeerPublicKey.ToHex()}"); 
+                        var messageTypes = toSend.Messages.Select(m =>
+                            m.MessageCase != NetworkMessage.MessageOneofCase.ConsensusMessage
+                                ? m.MessageCase.ToString()
+                                : m.ConsensusMessage.PrettyTypeString()
+                        );
+                        Logger.LogTrace($"Messages types: {string.Join("; ", messageTypes)}");
+                        foreach (var message in toSend.Messages)
+                        {
+                            MessageCounter
+                                .WithLabels(PeerPublicKey.ToHex(), message.MessageCase.ToString())
+                                .Inc();
+                            MessageBytesCounter
+                                .WithLabels(PeerPublicKey.ToHex(), message.MessageCase.ToString())
+                                .Inc(message.CalculateSize());
+                        }
+
+                        _hubConnector.Send(PeerPublicKey, megaBatchBytes);
+                        _eraMsgCounter += 1;
                     }
-
-                    _hubConnector.Send(PeerPublicKey, megaBatchBytes);
-                    _eraMsgCounter += 1;
+                    
+                    var toSleep = Math.Clamp(250 - (long) (TimeUtils.CurrentTimeMillis() - now), 1, 1000);
+                    Thread.Sleep(TimeSpan.FromMilliseconds(toSleep));
```

Comment:
Are we need this sleep? We have a Wait several lines ago

Reply:
I am not sure, it was already present so did not remove. Should I remove it and test?

Reply:
ok,  let's leave it for now,  will remove it later